src/onnx/onnx.cpp

```diff
@@ -1989,6 +1990,47 @@ struct onnx_parser
         return l0;
     }
 
+    instruction_ref
+    parse_range(const std::string&, const node_info&, std::vector<instruction_ref> args)
+    {
+
+        auto start_arg = args[0]->eval();
+        check_arg_empty(start_arg, "PARSE_RANGE: start arg dynamic shape is not supported");
+        auto limit_arg = args[1]->eval();
+        check_arg_empty(limit_arg, "PARSE_RANGE: limit arg dynamic shape is not supported");
+        auto delta_arg = args[2]->eval();
+        check_arg_empty(delta_arg, "PARSE_RANGE: delta arg dynamic shape is not supported");
+
+        assert(args[0]->get_shape().elements() == args[1]->get_shape().elements() ==
+               args[2]->get_shape().elements() == 1);
```

Comment:
Actually, this does not do what is expected. `args[0]->get_shape().elements() == args[1]->get_shape().elements()` will return a boolean. It should be `args[0]->get_shape().elements() == 1 && args[1]->get_shape().elements() == 1 && args[2]->get_shape().elements() == 1`.

Reply:
fixed